core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala

```diff
@@ -90,7 +92,15 @@ class BaseColumnFormatRelation(
   @transient protected lazy val region = Misc.getRegionForTable(resolvedName,
     true).asInstanceOf[PartitionedRegion]
 
-  override lazy val numPartitions: Int = region.getTotalNumberOfBuckets
+  override lazy val numPartitions: Int = {
+//    val region = Misc.getRegionForTable(resolvedName, true).
+//        asInstanceOf[PartitionedRegion]
+//    region.getTotalNumberOfBuckets
+    val numCores = Runtime.getRuntime.availableProcessors()
```

Comment:
This is evaluated at driver node. We need to consider sever nodes. Driver node num processors is not useful to us. Can you please see SchedulerBackend.defaultParallelism. That  takes total cores for slaves into consideration.
Catch however is spark.default.parallelism gets priority and if somebody configures bad we will suffer.